src/allmydata/scripts/magic_folder_cli.py

```diff
@@ -0,0 +1,198 @@
+
+import os, json
+from cStringIO import StringIO
+from twisted.python import usage
+
+from .common import BaseOptions, BasedirOptions, get_aliases
+from .cli import MakeDirectoryOptions, ListOptions, LnOptions
+import tahoe_ls, tahoe_mv
+from allmydata.util import fileutil
+
+
+INVITE_SEPERATOR = "~"
+
+class CreateOptions(BasedirOptions):
+    nickname = None
+    localdir = None
+    synopsis = "MAGIC_ALIAS: [NICKNAME LOCALDIR]"
+    def parseArgs(self, alias, nickname=None, localdir=None):
+        BasedirOptions.parseArgs(self)
+        self.alias = alias
+        self.nickname = nickname
+        self.localdir = localdir
+        if self.nickname and not self.localdir:
+            raise usage.UsageError("must provide both")
+        node_url_file = os.path.join(self['node-directory'], "node.url")
+        self['node-url'] = open(node_url_file, "r").read().strip()
+
+def create(options):
+    from allmydata.scripts import tahoe_add_alias
+    rc = tahoe_add_alias.create_alias(options)
+
+    if options.nickname is not None:
+        invite_options = InviteOptions()
+        invite_options.aliases = get_aliases(options['node-directory'])
+        invite_options["node-url"] = options["node-url"]
+        invite_options["node-directory"] = options["node-directory"]
+        invite_options.alias = options.alias
+        invite_options.nickname = options.nickname
+        invite_options.stdin = StringIO("")
+        invite_options.stdout = StringIO()
+        invite_options.strerr = StringIO()
+        rc = invite(invite_options)
+        if rc != 0:
+            print >>options.stderr, "magic-folder: failed to invite after create\n"
+            return -1
+        invite_code = invite_options.stdout.getvalue().strip()
```

Comment:
The output of invite should include instructions as well as the actual code (so maybe we need to just take the first line).